date_range/models/date_range_type.py

```diff
@@ -25,3 +27,19 @@ def _default_company(self):
     _sql_constraints = [
         ('date_range_type_uniq', 'unique (name,company_id)',
          'A date range type must be unique per company !')]
+
+    @api.constrains('company_id')
+    def _check_company_id(self):
+        if not self.env.context.get('bypass_company_validation', False):
+            for rec in self:
+                if not rec.company_id:
+                    continue
+                field = self.env['date.range'].search(
```

Comment:
@mreficent  you could maybe avoid a search into a loop  by using a `read_group` (not bloking)